syft/pkg/cataloger/common/cpe/sort_by_specificity.go

```diff
@@ -19,11 +20,30 @@ func (c BySpecificity) Less(i, j int) bool {
 	jScore := weightedCountForSpecifiedFields(c[j])
 
 	if iScore == jScore {
+		if countFieldLength(c[i]) == countFieldLength(c[j]) {
+			// lower index count preferred
+			return dashIndex(c[i]) > dashIndex(c[j])
+		}
+
 		return countFieldLength(c[i]) > countFieldLength(c[j])
 	}
 	return iScore > jScore
 }
 
+func dashIndex(cpe wfn.Attributes) int {
```

Comment:
A comment here as to why this is here would go a long way --It isn't obvious that this is here purely to stabilize the sort

Reply:
comment added here. I'm also open to talk about this being a pretty fragile assumption. It was a patch that was very target at the alpine example I listed in the PR. I'm still running some other images to see where/how this might break and if I can find a larger and more general solution.

Reply:
agreed --it's always worth a little extra time up front to attempt and avoid solving the same problem twice!

Reply:
I'm just mostly confused why are we not just text sorting? We could sort based on each field split based on `:` or somesuch

Reply:
yea there was some conversation on using 
```
return c[i].BindToFmtString() > c[j].BindToFmtString()
```

I'll check that rather than sorting based on random `-`